File: 2023/1/calibration.py
from re import findall

f = open('input.txt')
lines = f.readlines()
sum = 0
for line in lines:
  digits = list(map(int, findall(r'\d', line)))
  number = digits[0] * 10 + digits[-1]
  sum += number

print(sum)

File: 2023/1/calibration.scala
import scala.io.Source

@main def calibration() = 
  val bufferedSource = Source.fromFile("input.txt")
  val sum = bufferedSource.getLines.foldLeft(0) { (acc, line) =>
    val matches = """\d""".r.findAllMatchIn(line).toList
    val ints = matches.map(m => m.matched.toInt)
    acc + ints.head * 10 + ints.last
  }
  println(sum)

  bufferedSource.close